### ISSUE TITLE
No cursor theme/size with 2.7 snap package

### ISSUE THREAD

**Issue by elcste** [keepassxreboot/keepassxc#7622]
## Overview
Just what you want: another snap bug 😱  With version 2.7 install as a snap, KeepassXC doesn't follow the system cursor theme and size. The lack of theme us ugly, and maybe confusing for some of the non-arrow cursors, but the lack of size is an accessibility issue. Even those of us with normal eyesight may want a larger cursor on a hiDPI screen.

Tested with Ubuntu (GNOME) 21.10 and Jammy (dev release that will be 22.04 LTS).

## Steps to Reproduce
1. It's easier to see by making your cursor huge: Settings > Accessibility > Cursor Size.
2. Restart the app just in case.
3. Move the cursor in and out of the KeepassXC window and see that it doesn't follow the chosen large size. The theme is also an ugly fallback theme, although the arrow isn't too different from the default Ubuntu arrow. There is a bigger difference with the waiting cursors and others.

## Expected Behavior
In the snap previous to version 2.7, the cursor matched system theme and size.

## Actual Behavior
The cursor uses a fallback theme and is always small.

## Context
[NOTE]: # ( Give us any additional information you may have. )
I'm not a snap expert by any means, but I see there were [major changes](https://github.com/keepassxreboot/keepassxc/commit/ab007c48158f8d12159e93e0df2e98c1b81f61b1) to the snapcraft.yml, including removing a section with the comment "plugs for theming, font settings, **cursor** and to use gtk3 file chooser" and switching to the `kde-neon` extension.

Qalculate-Qt is another Qt snap that uses the `kde-neon` extension and has cursor theme/size working for me. Looking at the [snapcraft.yml](https://github.com/Qalculate/qalculate-qt/blob/main/snap/snapcraft.yaml), one difference I see is `plugs:` including `gsettings`. I don't want to suggest this is the solution, but it may be worth a try.

[NOTE]: # ( Paste debug info from Help → About here )
KeePassXC - 2.7.0
Revision: 20c9ac1

[NOTE]: # ( Pick choices based on your environment )
Operating System: Linux
Desktop Env: Gnome
Windowing System: both X11 and Wayland

**Comment by pbwn** [comment]
i have this issue on ubuntu 18.04 and keepass 2.7.0. My system has blink rate set to 0 but the cursor ignores this. It's the only app i use that ignores the blink rate 

ps, i make great use of keepass, thanks for it

**Comment by pkorba** [comment]
Same problem here on Kubuntu 20.04. Version 2.7.0 also doesn't respect system fonts anymore.

**Comment by droidmonkey** [comment]
This is a snap problem in general, nothing we can do about it. Trust me, we have been dealing with this problem for YEARS.

**Comment by elcste** [comment]
No disrespect intended, but I'm surprised you say that, because until this version the KeepassXC snap did this perfectly for as long as I have used it! I agree it is a snap flaw, as this has not always been true for other snaps.

For future reference of people finding this issue when searching, I am having luck setting `XCURSOR_SIZE=48 XCURSOR_THEME=Yaru` before running it under Sway. Easily done with a wrapper script. But it doesn't work for me under GNOME.

**Comment by droidmonkey** [comment]
With this new version of the snap I converted over to the kde-neon extension... which is the snap endorsed way to build and deploy a qt application. Quite literally I stripped away a ton of workarounds from our snapcraft file. Everything works great in Ubuntu 20.04. Anywhere else? Complete shit. I literally give up, about ready to delete the snap entirely.

https://snapcraft.io/docs/qt5-kde-applications
https://snapcraft.io/docs/kde-neon-extension

**Comment by elcste** [comment]
I appreciate your hard work and feel your frustration. I think you're getting a lot of user feedback like this because your app and snap quality has been consistently so high! I saw your post on the snapcraft forum on the open/save bug and might tag along with this cosmetic stuff if you get some answers.

**Comment by droidmonkey** [comment]
Yes I think these are all related, I can't go back to the original way we did snaps because it doesn't let me use the modern Qt and packages (like Botan). I am stuck in a dumb Ubuntu rut.

**Comment by townsend2010** [comment]
> Everything works great in Ubuntu 20.04.

I will add I'm using Ubuntu 20.04 and see the no theming issue :slightly_frowning_face:

**Comment by q-wertz** [comment]
Maybe a stupid idea, but don't you need the `gsettings` plug to access the information on the curser size?

EDIT: Ah sorry for the noise. was already in the first post.

**Comment by q-wertz** [comment]
Maybe related: Output of `snappy-debug` which is related to keepassxc
```text
= AppArmor =
Time: Mar 29 11:26:23
Log: apparmor="DENIED" operation="open" profile="snap.keepassxc.keepassxc" name="/snap/core20/1405/usr/lib/x86_64-linux-gnu/libdl-2.31.so" pid=285103 comm="keepassxc" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0
File: /snap/core20/1405/usr/lib/x86_64-linux-gnu/libdl-2.31.so (read)
Suggestion:
* adjust program to read necessary files from $SNAP, $SNAP_DATA, $SNAP_COMMON, $SNAP_USER_DATA or $SNAP_USER_COMMON

= AppArmor =
Time: Mar 29 11:26:23
Log: apparmor="DENIED" operation="open" profile="snap.keepassxc.keepassxc" name="/snap/core20/1405/usr/lib/x86_64-linux-gnu/libgssapi_krb5.so.2.2" pid=285103 comm="keepassxc" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0
File: /snap/core20/1405/usr/lib/x86_64-linux-gnu/libgssapi_krb5.so.2.2 (read)
Suggestion:
* adjust program to read necessary files from $SNAP, $SNAP_DATA, $SNAP_COMMON, $SNAP_USER_DATA or $SNAP_USER_COMMON

= AppArmor =
Time: Mar 29 11:26:23
Log: apparmor="DENIED" operation="open" profile="snap.keepassxc.keepassxc" name="/snap/core20/1405/usr/lib/x86_64-linux-gnu/libdbus-1.so.3.19.11" pid=285103 comm="keepassxc" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0
File: /snap/core20/1405/usr/lib/x86_64-linux-gnu/libdbus-1.so.3.19.11 (read)
Suggestion:
* adjust program to read necessary files from $SNAP, $SNAP_DATA, $SNAP_COMMON, $SNAP_USER_DATA or $SNAP_USER_COMMON

= AppArmor =
Time: Mar 29 11:26:23
Log: apparmor="DENIED" operation="open" profile="snap.keepassxc.keepassxc" name="/snap/core20/1405/usr/lib/x86_64-linux-gnu/libsystemd.so.0.28.0" pid=285103 comm="keepassxc" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0
File: /snap/core20/1405/usr/lib/x86_64-linux-gnu/libsystemd.so.0.28.0 (read)
Suggestion:
* adjust program to read necessary files from $SNAP, $SNAP_DATA, $SNAP_COMMON, $SNAP_USER_DATA or $SNAP_USER_COMMON

= AppArmor =
Time: Mar 29 11:26:23
Log: apparmor="DENIED" operation="open" profile="snap.keepassxc.keepassxc" name="/snap/core20/1405/usr/lib/x86_64-linux-gnu/libzstd.so.1.4.4" pid=285103 comm="keepassxc" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0
File: /snap/core20/1405/usr/lib/x86_64-linux-gnu/libzstd.so.1.4.4 (read)
Suggestion:
* adjust program to read necessary files from $SNAP, $SNAP_DATA, $SNAP_COMMON, $SNAP_USER_DATA or $SNAP_USER_COMMON

= AppArmor =
Time: Mar 29 11:26:23
Log: apparmor="DENIED" operation="open" profile="snap.keepassxc.keepassxc" name="/snap/core20/1405/usr/lib/x86_64-linux-gnu/libglib-2.0.so.0.6400.6" pid=285103 comm="keepassxc" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0
File: /snap/core20/1405/usr/lib/x86_64-linux-gnu/libglib-2.0.so.0.6400.6 (read)
Suggestion:
* adjust program to read necessary files from $SNAP, $SNAP_DATA, $SNAP_COMMON, $SNAP_USER_DATA or $SNAP_USER_COMMON

= AppArmor =
Time: Mar 29 11:26:23
Log: apparmor="DENIED" operation="open" profile="snap.keepassxc.keepassxc" name="/snap/core20/1405/usr/lib/x86_64-linux-gnu/libkrb5.so.3.3" pid=285103 comm="keepassxc" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0
File: /snap/core20/1405/usr/lib/x86_64-linux-gnu/libkrb5.so.3.3 (read)
Suggestion:
* adjust program to read necessary files from $SNAP, $SNAP_DATA, $SNAP_COMMON, $SNAP_USER_DATA or $SNAP_USER_COMMON

= AppArmor =
Time: Mar 29 11:26:23
Log: apparmor="DENIED" operation="open" profile="snap.keepassxc.keepassxc" name="/snap/core20/1405/usr/lib/x86_64-linux-gnu/libk5crypto.so.3.1" pid=285103 comm="keepassxc" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0
File: /snap/core20/1405/usr/lib/x86_64-linux-gnu/libk5crypto.so.3.1 (read)
Suggestion:
* adjust program to read necessary files from $SNAP, $SNAP_DATA, $SNAP_COMMON, $SNAP_USER_DATA or $SNAP_USER_COMMON

= AppArmor =
Time: Mar 29 11:26:23
Log: apparmor="DENIED" operation="open" profile="snap.keepassxc.keepassxc" name="/snap/core20/1405/usr/lib/x86_64-linux-gnu/libcom_err.so.2.1" pid=285103 comm="keepassxc" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0
File: /snap/core20/1405/usr/lib/x86_64-linux-gnu/libcom_err.so.2.1 (read)
Suggestion:
* adjust program to read necessary files from $SNAP, $SNAP_DATA, $SNAP_COMMON, $SNAP_USER_DATA or $SNAP_USER_COMMON
```

**Comment by elcste** [comment]
I just noticed that the cursor theme/size is working correctly with the 2.7.1 snap, at least on Ubuntu 21.10/X11. I can test 22.04/Wayland at work tomorrow.

**Comment by elcste** [comment]
Yes, with the current version 2.7.1 snap, I'm seeing the correct cursor size and theme on Ubuntu Jammy (future 22.04) GNOME/Wayland, and Sway as well.

I don't see anything in the changelog about the snap, or changes to the snapcraft.yaml. If nothing was specifically done, maybe it was fixed by the Flatpak improvements (or maybe it was the updated "core" snap, which I got at the same time).

As someone else mentioned, the font does not follow the system theme, which is still the case as I can see, but that's not as unexpected for me.
